layout_parser/src/bin/cli.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::Deserialize;
+use std::error::Error;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+
+use structopt::StructOpt;
+
+use layout_parser::Layout;
+
+#[derive(StructOpt)]
+struct Opt {
+    /// input layout file
+    #[structopt(long, parse(from_os_str))]
+    path: PathBuf,
+}
+
+fn read_layout_from_file<P: AsRef<Path>>(path: P) -> Result<Layout, Box<dyn Error>> {
+    // Open the file in read-only mode with buffer.
+    let file = File::open(path)?;
+    let reader = BufReader::new(file);
+
+    // Read the JSON contents of the file as an instance of `Layout`.
+    // need to use the disabled recursion limit for large layouts (e.g. external_transition)
+
+    let mut deserializer = serde_json::Deserializer::from_reader(reader);
+    deserializer.disable_recursion_limit();
+    let deserializer = serde_stacker::Deserializer::new(&mut deserializer);
+    let u = Layout::deserialize(deserializer)?;
```

Comment:
it looks like this just deserializes the layout from JSON, but it doesn't parse it into the bin_prot types unless I'm missing something? would be good to get that parsing in, but happy to approve this and that can be done in a follow up

Reply:
Follow https://github.com/MinaProtocol/mina/issues/8718 for developments on this. It would be great if we merge this after implementing the ability to generate Rust types or conclude if it isn't possible after improvements in the layout JSON.

Reply:
@creativcoder sure sounds good. This PR just includes the ability to deserialize the layouts into a type tree but we really need to decide what we want to do with that.